### PR TITLE
Feature: update registration form with submit actions

### PR DESCRIPTION
**Pull request by ShellyXueHan**

### CODE REVIEW

File: src/actionCreators/index.js [hunk 1]
```diff
@@ -41,18 +53,46 @@ export const authorize = (ssoGroup, email) => {
...
+
+export const updateUser = (userId, userProfile) => {
+  return dispatch => {
+    dispatch(updateUserStart());
+    axi
+      .put(
+        API.UPDATE_SSO_USER(userId),
+        { ...userProfile },
+        {
+          headers: {
```
Comment (patricksimonian) [inline src/actionCreators/index.js]: do the request headers ever change? Maybe we can make a function that creates it, a constant for it, and or have it preconfiged within the axios instance

File: src/actionCreators/index.js [hunk 2]
```diff
@@ -41,18 +53,46 @@ export const authorize = (ssoGroup, email) => {
...
   };
 };
+
+export const updateUser = (userId, userProfile) => {
+  return dispatch => {
+    dispatch(updateUserStart());
+    axi
+      .put(
+        API.UPDATE_SSO_USER(userId),
+        { ...userProfile },
```
Comment (patricksimonian) [inline src/actionCreators/index.js]: don't need to spread user profile at this point i don't think

File: src/actionCreators/index.js [hunk 3]
```diff
@@ -41,18 +53,46 @@ export const authorize = (ssoGroup, email) => {
...
+          headers: {
+            Accept: 'application/json',
+          },
+        }
+      )
+      .then(res => {
+        return dispatch(updateUserSuccess());
+      })
+      .catch(err => {
+        const errMsg = 'Fail to register your account, please try again.';
```
Comment (patricksimonian) [inline src/actionCreators/index.js:83]: put in constants?

File: src/components/App.js [hunk 4]
```diff
@@ -24,9 +24,12 @@ import { bindActionCreators } from 'redux';
 import { Route, Switch } from 'react-router-dom';
 import { authenticateFailed, authenticateSuccess } from '../actions';
 import implicitAuthManager from '../auth';
-import Home from '../containers/Home';
-import Registration from '../containers/Registration';
-import RocketChat from '../containers/RocketChat';
+import { Confirmation, Home, Registration, RocketChat, Rejection } from '../containers';
+// import Confirmation from '../containers/Confirmation';
```
Comment (patricksimonian) [inline src/components/App.js]: remove commented lines

File: src/containers/Confirmation.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    return (
+      <div>
+        <h1>Welcome back,</h1>
+        <h2>{this.props.location.search}</h2>
+      </div>
+    );
+  }
+}
+
+const mapStateToProps = state => {
```
Comment (patricksimonian) [inline src/containers/Confirmation.js:37]: does this component need to be connected?

File: src/containers/Home.js [hunk 6]
```diff
@@ -28,21 +28,18 @@ class Home extends Component {
   static displayName = '[Component Home]';
 
   render() {
-    // TODO: combine isAuthorized and authorizationStarted:
     let authorizedRedirect = null;
-    if (this.props.isAuthorized) {
+    if (this.props.isAuthorized === 2) {
```
Comment (patricksimonian) [inline src/containers/Home.js]: nice i like this!

File: src/containers/Confirmation.js (hunk 5 above)
Comment (ShellyXueHan) [reply src/containers/Confirmation.js:37]: will need it in next pr

**Comment by ShellyXueHan** [comment]
@patricksimonian ready for review!

**Review by patricksimonian — APPROVED** [review]

**Pull request merged**